rust/ql/lib/codeql/rust/elements/LogicalOperation.qll

```diff
@@ -1,33 +1,46 @@
 private import codeql.rust.elements.Expr
 private import codeql.rust.elements.BinaryExpr
 private import codeql.rust.elements.PrefixExpr
+private import codeql.rust.elements.Operation
 
-abstract private class LogicalOperationImpl extends Expr {
-  abstract Expr getAnOperand();
-}
+/**
+ * A logical operation, such as `&&`, `||` or `!`.
+ */
+abstract private class LogicalOperationImpl extends Operation { }
 
 final class LogicalOperation = LogicalOperationImpl;
 
-abstract private class BinaryLogicalOperationImpl extends BinaryExpr, LogicalOperationImpl {
-  override Expr getAnOperand() { result = [this.getLhs(), this.getRhs()] }
-}
+/**
+ * A binary logical operation, such as `&&` or `||`.
+ */
+abstract private class BinaryLogicalOperationImpl extends BinaryExpr, LogicalOperationImpl { }
 
 final class BinaryLogicalOperation = BinaryLogicalOperationImpl;
 
+/**
+ * The logical and operation, `&&`.
```

Comment:
Could we do something to distinguish that "and" not a normal English occurrence of the word? Maybe wrap it in quotes like this:
```suggestion
 * The logical "and" operation, `&&`.
```

Reply:
I agree that's an improvement. :+1: